Tella/Data/Database/UwaziDatabase.swift

```diff
@@ -183,15 +198,207 @@ extension TellaDataBase: UwaziServerLanguageProtocol {
+            
+            return try responseDict.compactMap({ dict in
+                let entityInstance = try dict.decode(UwaziEntityInstance.self)
+                
+                
+                let server = try getUwaziServer(serverId: entityInstance.serverId)
+                entityInstance.server = server
+                
+                let collectedTemplate = try getUwaziTemplate(templateId: entityInstance.templateId)
+                entityInstance.collectedTemplate = collectedTemplate
+                
+                entityInstance.files = getVaultFiles(instanceId: entityInstance.id)
+                return entityInstance
+            })
+        } catch let error {
+            debugLog(error)
+            return []
+        }
+    }
+    
+    func getUwaziEntityInstance(entityId:Int) -> UwaziEntityInstance? {
+        
+        do {
+            let entityInstanceCondition = [KeyValue(key: D.cId, value: entityId)]
+            let responseDict = try statementBuilder.selectQuery(tableName: D.tUwaziEntityInstances,
+                                                                andCondition: entityInstanceCondition)
+            
+            guard let entityInstance = try responseDict.first?.decode(UwaziEntityInstance.self) else  {return nil}
+            
+            let server = try getUwaziServer(serverId: entityInstance.serverId)
+            entityInstance.server = server
+            
+            let collectedTemplate = try getUwaziTemplate(templateId: entityInstance.templateId)
+            entityInstance.collectedTemplate = collectedTemplate
+            
+            entityInstance.files = getVaultFiles(instanceId: entityInstance.id)
+            return entityInstance
+
+        } catch let error {
+            debugLog(error)
+            return nil
+        }
+    }
+    
+    func getVaultFiles(instanceId:Int?) -> [UwaziEntityInstanceFile] {
+        
+        do {
+            guard let instanceId else {return [] }
+            let instanceCondition = [KeyValue(key: D.cUwaziEntityInstanceId, value: instanceId)]
+            let responseDict = try statementBuilder.selectQuery(tableName: D.tUwaziEntityInstanceVaultFile,
+                                                                andCondition: instanceCondition)
+            let entityInstances = try responseDict.decode(UwaziEntityInstanceFile.self)
+            return entityInstances
+            
+        } catch {
+            return []
+        }
+    }
+    
+    func deleteEntityInstance(entityId : Int) -> Result<Bool,Error> {
+        
+        do {
+            
+            guard let entity = self.getUwaziEntityInstance(entityId: entityId) else {
+                return .failure(RuntimeError("No Entity is selected"))
+            }
+            
+            try deleteEntityInstanceFiles(entityIds: [entityId])
+            
+            let entityCondition = [KeyValue(key: D.cId, value: entity.id as Any)]
+            
```

Comment:
Why you are using as `Any` and the `entity.id` is an Int type. Maybe change it to `value: entity.id as Int?` ?